src/base/bittorrent/torrentinfo.cpp

```diff
@@ -246,6 +246,28 @@ QVector<int> TorrentInfo::fileIndicesForPiece(int pieceIndex) const
     return res;
 }
 
+QByteArray TorrentInfo::hashForPiece(const int pieceIndex) const
+{
+    if (!isValid() || (pieceIndex < 0) || (pieceIndex >= piecesCount()))
+        return {};
+    return { m_nativeInfo->hash_for_piece_ptr(pieceIndex), libtorrent::sha1_hash::size };
+}
+
+QVector<QByteArray> TorrentInfo::piecesHashes() const
+{
+    if (!isValid())
+        return {};
+
+    const int count = piecesCount();
+    QVector<QByteArray> hashes;
+    hashes.reserve(count);
+
+    for (int i = 0; i < count; ++i)
+        hashes.append(hashForPiece(i));
```

Comment:
I don't know. Extracting a single function call into its own method, protecting it with 3 checks just to repeat this in a loop is simply does not make sense, as to me. Is `hashForPiece()` needed at all?